stellargraph/core/graph.py

```diff
@@ -196,41 +196,59 @@ def has_node(self, node: Any) -> bool:
         """
         raise NotImplementedError
 
-    def neighbors(self, node: Any) -> Iterable[Any]:
+    def neighbors(
+        self, node: Any, include_edge_weight=False, edge_types=None
+    ) -> Iterable[Any]:
         """
         Obtains the collection of neighbouring nodes connected
         to the given node.
 
         Args:
             node (any): The node in question.
+            include_edge_weight (bool, default False): If True, each neighbour in the
+                output is a named tuple with fields `node` (the node ID) and `weight` (the edge weight)
+            edge_types (list of hashable, optional): If provided, only traverse the graph
+                via the provided edge types when collecting neighbours.
 
         Returns:
             iterable: The neighbouring nodes.
         """
         raise NotImplementedError
```

Comment:
Github isn't noticing it as a formal merge conflict, but I think there's a conceptual one here: on `develop`, these all now call `self._graph.neighbors(...)` etc., and so they'll need to be updated to pass the arguments through:

https://github.com/stellargraph/stellargraph/blob/69dac010e106c93a6036638dc2bf2abfb82e84a9/stellargraph/core/graph.py#L197

- [ ] Similarly for all of the other ones